### PR TITLE
Prepare http body handling

### PR DESCRIPTION
**Pull request by brummer-simon**

### CODE REVIEW

File: source/extensions/filters/http/grpc_http1_reverse_bridge_transcoder/filter.cc [hunk 1]
```diff
@@ -58,7 +64,10 @@ Grpc::Status::GrpcStatus grpcStatusFromHttpStatus(uint64_t http_status) {
 Filter::Filter(Api::Api& api, std::string proto_descriptor_path, std::string service_name)
     : transcoder_{}, grpc_sessions_{} {
   auto const status = transcoder_.init(api, proto_descriptor_path, service_name);
-  assert(status.ok());
+  if (!status.ok()) {
+    ENVOY_LOG(critical, "Failed to intialize transcoder. Error was: {}", status.message());
+    assert(false);
```
Comment (aledzu) [inline source/extensions/filters/http/grpc_http1_reverse_bridge_transcoder/filter.cc]: What will happen in a release build?

File: source/extensions/filters/http/grpc_http1_reverse_bridge_transcoder/filter.cc [hunk 2]
```diff
@@ -100,67 +109,88 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap& headers,
...
     ENVOY_STREAM_LOG(error,
                      "Unable to create session. Send gRPC error message downstream. Error was: {}",
-                     *decoder_callbacks_, session_status.status().message());
+                     *decoder_callbacks_, session_or.status().message());
 
     respondWithGrpcError(*decoder_callbacks_, Errors::InternalError);
     return Http::FilterHeadersStatus::StopIteration;
   }
-  auto* const session = *session_status;
+  auto* const session = *session_or;
```
Comment (aledzu) [inline source/extensions/filters/http/grpc_http1_reverse_bridge_transcoder/filter.cc]: auto const* const? :)

Comment (brummer-simon) [reply source/extensions/filters/http/grpc_http1_reverse_bridge_transcoder/filter.cc]: good catch

File: source/extensions/filters/http/grpc_http1_reverse_bridge_transcoder/filter.cc (hunk 1 above)
Comment (brummer-simon) [reply source/extensions/filters/http/grpc_http1_reverse_bridge_transcoder/filter.cc]: Good catch.

**Review by aledzu — APPROVED** [review]

**Pull request merged**